identity/graphql.go

```diff
@@ -0,0 +1,353 @@
+	return i.identity.Sig
+}
+
+// ClaimArgs are the arguments for a GraphQL claim query.
+type ClaimArgs struct {
+	Issuer    *string
+	Holder    *string
+	Claim     *string
+	Signature *string
+	ID        *string
+}
+
+// Claim is a GraphQL resolver function which retrieves object Claims from the
+// SQLite3 index using either Claim ID,Issuer,holder,Claim or Signature and loads the
+// associated META objects from the META store.
+func (r *Resolver) Claim(args ClaimArgs) ([]*ClaimResolver, error) {
+	var rows *sql.Rows
+	var err error
+	switch {
+	case args.Issuer != nil:
+		rows, err = r.db.Query("SELECT object_id FROM claim WHERE issuer = ?", *args.Issuer)
+	case args.Holder != nil:
+		rows, err = r.db.Query("SELECT object_id FROM claim WHERE holder = ?", *args.Holder)
+	case args.Claim != nil:
+		rows, err = r.db.Query("SELECT object_id FROM claim WHERE holder = ?", *args.Claim)
+	case args.Signature != nil:
+		rows, err = r.db.Query("SELECT object_id FROM claim WHERE signature = ?", *args.Signature)
+	case args.ID != nil:
+		rows, err = r.db.Query("SELECT object_id FROM claim WHERE id = ?", *args.ID)
+	default:
+		return nil, errors.New("missing issuer,holder,claim,signature or id argument")
+	}
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var resolvers []*ClaimResolver
+	for rows.Next() {
+		var objectID string
+		if err := rows.Scan(&objectID); err != nil {
+			return nil, err
+		}
+		objCid, err := cid.Parse(objectID)
+		if err != nil {
+			return nil, err
+		}
+		obj, err := r.store.Get(objCid)
+		if err != nil {
+			return nil, err
+		}
+		var claim Claim
+		if err := obj.Decode(&claim); err != nil {
+			return nil, err
+		}
+		resolvers = append(resolvers, &ClaimResolver{objectID, &claim})
+	}
+	if err := rows.Err(); err != nil {
+		return nil, err
+	}
+	return resolvers, nil
+}
+
+// ClaimResolver defines GraphQL resolver functions for Claim fields.
+type ClaimResolver struct {
+	cid   string
+	claim *Claim
+}
+
+// Cid resolver
+func (c *ClaimResolver) Cid() string {
+	return c.cid
+}
+
+// Issuer resolver
+func (c *ClaimResolver) Issuer() string {
+	return c.claim.Issuer
+}
+
+// Holder resolver
+func (c *ClaimResolver) Holder() string {
+	return c.claim.Holder
+}
+
+// Claim resolver
+func (c *ClaimResolver) Claim() string {
+	return c.claim.Claim
+}
+
+// Signature resolver
+func (c *ClaimResolver) Signature() string {
+	return c.claim.Signature
+}
+
+// ID resolver
+func (c *ClaimResolver) ID() string {
+	return c.claim.ID
+}
+
+// CreateIDArgs are the arguments for a GraphQL CreateID mutation.
+type CreateIDArgs struct {
+	Username  *string
+	Owner     *string
+	Signature *string
+}
+
+// CreateID is a GraphQL resolver function which create ID object, store and
+// index it on SQLite3 index.
+func (r *Resolver) CreateIdentity(args *CreateIDArgs) (*IdentityResolver, error) {
+
+	metaid, err := NewIdentity(*args.Username, common.HexToAddress(*args.Owner), common.FromHex(*args.Signature))
```

Comment:
Check that the fields are all set to avoid a panic (or make them all required in the schema so that they are `string` rather than `*string`).